internal/agent/agent_test.go

```diff
@@ -1,87 +1,35 @@
-package agent
+package agent_test
```

Comment:
We need to avoid a circular dependency, we can test the agent without internal knowledge here.

Reply:
if we moved the test into test/integration/agent would that help and consume your harness?

Reply:
That would also work, I find it nicer to keep test and tested code closer while we can. But we could change that later if we think that's better.